File: benchmark/c/cpw/double_factorial_1/double_factorial_1.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



unsigned int double_factorial_1 ( unsigned int n ) ;
unsigned int double_factorial_1 ( unsigned int n ) {
  int res = 1;
  for ( int i = n;
  i >= 0;
  i = i - 2 ) {
    if ( i == 0 || i == 1 ) return res;
    else res *= i;
  }
}

File: translation_result/else_bms/success/c/cpw/double_factorial_1/gpt4/CAPR/double_factorial_1.rs

fn double_factorial_1(n: u32) -> u32 {
    let mut res: u32 = 1;
    let mut i = n;

    while i > 1 {
        res = res.wrapping_mul(i);
        i = i.wrapping_sub(2);
    }

    res
}
